model/database/cosmosdb-metrics.yaml

```diff
@@ -7,6 +7,12 @@ groups:
     unit: "{request_unit}"
     stability: experimental
     extends: attributes.db.cosmosdb.minimal
+    attributes:
+      - ref: db.cosmosdb.regions_contacted
```

Comment:
creating an array per each call could be expensive - did we consider this? 

One way to approach it could be to add some config option to turn the attribute off in the future,

For now I think it's important to leave a room for it - in particular change `requirement_level` to recommended  instead of `conditionally_required`

Reply:
creating an array per each call could be expensive - did we consider performance aspect of this attribute? => we already have it in `Set`, we really don't have to do much to convert it to `array` or `string` and I am fine with anything.

@kirankumarkolli @jcocchi what do you think about making it opt-in? if customer is on multi-region set up, only then this dimension would be required.

Reply:
I don't have a preference on string vs. array type, but I think having an opt in for `regions_contacted` might be overly complicated. What scenarios would we recommend vs. not recommend it? What would the default be? 

For cases we would recommend not having it, the impact is low. If a customer only has one region then the list would always be a single value. For two region set ups which is common, there would only be 4 total possible values here. For high impact cases with a lot of permutations, the attribute is very useful and we would recommend they capture it

Reply:
I agree, we can have it always by default.

Reply:
unresolving this. It's still conditionally_required. The recommendation is to make it `recommended: if available` to make it future-proof (allow opting out)